src/reporters/remote_reporter.js

```diff
@@ -65,11 +65,20 @@ export default class RemoteReporter {
         } else {
             this._metrics.reporterSuccess.increment(response.numSpans);
         }
+
+        if (callback) {
+            callback();
+        }
     }
 
     close(callback: ?Function): void {
-        this._sender.close(callback);
+        this._sender.flush();
+        this._sender.close();
         clearInterval(this._intervalHandle);
+
```

Comment:
I meant you should stop the timer first before flushing/closing the sender, as otherwise another flush might be called from the timer before you stop it.